### PR TITLE
add SklearnCallback as integration with sklearn metrics

### PR DESCRIPTION
**Pull request by y-ksenia**
## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [x] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Did you run [colab minimal CI/CD](https://colab.research.google.com/drive/1JCGTVvWlrIsLXMPRRRSWiAstSLic4nbA) with `latest` and `minimal` requirements?

## Description

Work on #1183

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [ ] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [ ] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [x] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [x] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

### CODE REVIEW

File: catalyst/callbacks/metrics/sklearn.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from catalyst.metrics import FunctionalBatchMetric
+
+
+class SklearnCallback(FunctionalBatchMetricCallback):
+    """@TODO: Docs."""
+
+    def __init__(
+        self,
+        keys: Mapping[str, str],
+        metric_fn: Callable,
```
Comment (Scitator) [inline catalyst/callbacks/metrics/sklearn.py]: could we make it [Callable, str]?
and use `metric_fn=sklearn.metrics.__dict__[metric_fn]` in str case? (we also have to check, that metric_fn isin dict.keys)(, btw)

File: catalyst/callbacks/metrics/sklearn.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from typing import Callable, Dict, Mapping
+
+import torch
+
+from catalyst.callbacks.metric import FunctionalBatchMetricCallback
+from catalyst.core.runner import IRunner
+from catalyst.metrics import FunctionalBatchMetric
+
+
+class SklearnCallback(FunctionalBatchMetricCallback):
```
Comment (Scitator) [inline catalyst/callbacks/metrics/sklearn.py]: is it possible to also add loader-based sklearn metrics into the separate Callback?
we could not compute AUCs in per-batch mode

Comment (Scitator) [reply catalyst/callbacks/metrics/sklearn.py]: nevertheless, we could just filter the loader-based metrics during metric_fn check for now
to simplify the PR

**Review by Scitator — COMMENTED** [review]
could you please integrate the Callback to a [minimal example](https://github.com/catalyst-team/catalyst/blob/master/catalyst/tests/test_finetune2.py) to check the performance?

File: catalyst/callbacks/metrics/__init__.py [hunk 3]
```diff
@@ -29,3 +29,5 @@
     IOUCallback,
     TrevskyCallback,
 )
+
+from catalyst.callbacks.metrics.scikit_learn import SklearnCallback
```
Comment (Scitator) [inline catalyst/callbacks/metrics/__init__.py]: let's import that callback only if `sklearn` is available in extensions
here is a good PR as example https://github.com/catalyst-team/catalyst/pull/1196

File: catalyst/callbacks/metrics/scikit_learn.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+class SklearnCallback(FunctionalBatchMetricCallback):
+    """
+
+    Args:
+        keys:
+        metric_fn:
+        metric_key:
+        log_on_batch:
+    """
+
```
Comment (Scitator) [inline catalyst/callbacks/metrics/scikit_learn.py]: could you please add docs to the metrics + to the official [API docs](https://github.com/catalyst-team/catalyst/blob/master/docs/api/metrics.rst)

Comment (y-ksenia) [reply catalyst/callbacks/metrics/scikit_learn.py]: WIP🙂

File: catalyst/callbacks/metrics/sklearn.py (hunk 1 above)
Comment (y-ksenia) [reply catalyst/callbacks/metrics/sklearn.py]: done

File: catalyst/callbacks/metrics/sklearn.py (hunk 2 above)
Comment (y-ksenia) [reply catalyst/callbacks/metrics/sklearn.py]: should I filter by name?.. or how?

File: catalyst/callbacks/metrics/__init__.py (hunk 3 above)
Comment (y-ksenia) [reply catalyst/callbacks/metrics/__init__.py]: done

File: catalyst/callbacks/metrics/scikit_learn.py (hunk 4 above)
Comment (y-ksenia) [reply catalyst/callbacks/metrics/scikit_learn.py]: done

**Comment by ditwoo** [comment]
@y-ksenia btw you can test `SklearnCallback` emulating experiment flow (without executing the whole experiment).

File: tests/catalyst/tests/test_finetune2.py [hunk 5]
```diff
@@ -115,6 +129,16 @@ def get_callbacks(self, stage: str):
...
+            callbacks["f1_score"] = dl.SklearnCallback(
+                keys={
+                    "y_pred": "labels",
+                    "y_true": "targets",
+                    "average": "micro",
+                    "zero_division": 1,
+                },
+                metric_fn="f1_score",
+                metric_key="f1_score",
+            )
```
Comment (Scitator) [inline tests/catalyst/tests/test_finetune2.py]: could you please add this example to the callbacks docs? it's very interesting, that you could also pass metrics **kwargs through the keys

**Review by Scitator — COMMENTED** [review]
could you please also update [Config API](https://github.com/catalyst-team/catalyst/tree/master/examples/mnist_stages) example with extra SklearnCallback?

File: catalyst/callbacks/metrics/scikit_learn.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                    metric_fn="f1_score",
+                    metric_key="f1_score",
+                )
+            ]
+        )
+
+    .. note::
+        Please follow the `minimal examples`_ sections for more use cases.
+
+        .. _`minimal examples`: https://github.com/catalyst-team/catalyst#minimal-examples
```
Comment (Scitator) [inline catalyst/callbacks/metrics/scikit_learn.py:104]: could please add a note, that `catalyst[ml]` required?

File: docs/api/callbacks.rst [hunk 7]
```diff
@@ -233,6 +233,20 @@ FunctionalMetricCallback
...
+    :exclude-members: __init__, on_experiment_start, on_stage_start, on_epoch_start, on_loader_start, on_batch_start, on_batch_end, on_loader_end, on_epoch_end, on_stage_end, on_experiment_end
+    :show-inheritance:
+
+SklearnLoaderCallback
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. autoclass:: catalyst.callbacks.metrics.scikit_learn.SklearnLoaderCallback
+    :members:
+    :exclude-members: __init__, on_experiment_start, on_stage_start, on_epoch_start, on_loader_start, on_batch_start, on_batch_end, on_loader_end, on_epoch_end, on_stage_end, on_experiment_end
+    :show-inheritance:
+
```
Comment (Scitator) [inline docs/api/callbacks.rst]: could you please bring them after `RecSys` to make everything in alphabetical order?

**Review by Scitator — DISMISSED** [review]
looks like only changelog.md update required :)

**Pull request merged**